### DISCUSSION TITLE
Kafka with Kraft post restart - re-generate clusterId

### DISCUSSION THREAD

**Discussion by shaitams** [strimzi/strimzi-kafka-operator#9109] [Q&A] [answered]
When i restart kafka/kraft cluster, the nodePools (controller,broker) generated new clusterId that doesn't match the clusterId on kafka crd and on the meta.properties file in the storage.

How to prevent this?

**Comment by scholzj** [comment]
First, you should share your version, resources, logs etc.

**Reply by scholzj** [reply]
You should also explain what exactly you mean with `When i restart kafka/kraft cluster`.

**Reply by shaitams** [reply]
Delete the kafka and the nodePools, and re-create them on the same namespace with the previous volumes

**Reply by scholzj** [reply]
That is not _restarting the cluster_. That is deleting the cluster. You should delete the storage and start with a new one.

**Reply by shaitams** [reply]
What about a case to move the kafka cluster to different k8s cluster, or on disaster recovery, or on k8s cluster issue?

**Reply by scholzj** [reply]
That might be something to address in the future. But in any case that is not restarting the cluster.

**Reply by shaitams** [reply]
There is a workaround to address this with initContainer script?

**Reply by scholzj** [reply]
I think the only workaround you can use is to restore the custom resources including their status sections that has the cluster ID.

**Reply by shaitams** [reply]
If I will move to work with zookeeper instead, it will be supported?

**Reply by scholzj** [reply]
Sorry, but what exactly? Recovering the Kafka cluster only from the PVCs?

**Reply by shaitams** [reply]
Yes

**Reply by scholzj** [reply]
Yeah, you can do that with ZooKeeper-based cluster. There is a section i the doc describing it.

**Reply by shaitams** [reply]
Can you send the link to the instructions?

**Reply by shaitams** [reply]
Ans, what are the disadvantages with zookeeper based cluster?

**Reply by scholzj** [reply, marked as answer]
I don't have the link at hand. But if you go to the docs and search for recovery, you should find it. The advantage of the ZooKeeper-based Kafka cluster is that it is production-ready. KRaft is not yet production ready.

**Reply by shaitams** [reply]
Thanks
